### PR TITLE
Feature/kdl create tokens page

### PR DESCRIPTION
**Pull request by marco-gattelli**

### CODE REVIEW

File: kdl-client/src/Constants/routes.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ enum ROUTE {
   SERVER = '/server/:serverId',
   SERVER_USERS = '/server/:serverId/users',
   USER_SSH_KEY = '/server/:serverId/user/ssh-key',
+  USER_API_TOKEN = '/server/:serverId/user/api-token',
```
Comment (marco-gattelli) [inline kdl-client/src/Constants/routes.ts:11]: @mgs95 please review the URL I choose for this page

File: kdl-client/graphql/schema.graphql [hunk 2]
```diff
@@ -14,6 +15,12 @@ type Mutation {
   updateProject(input: UpdateProjectInput!): Project!
 }
 
+type ApiToken {
+  label: String!
```
Comment (marco-gattelli) [inline kdl-client/graphql/schema.graphql:19]: @mgs95 shouldn't I include an `ID!` for the api token object?

Comment (mgs95) [reply kdl-client/graphql/schema.graphql:19]: You should, for deleting a token you should pass the ID to tell the serve which token to remove

File: kdl-client/graphql/schema.graphql [hunk 3]
```diff
@@ -3,6 +3,7 @@ type Query {
   projects: [Project!]!
   users: [User!]!
   sshKey: SSHKey!
+  apiTokens: [ApiToken!]!
```
Comment (mgs95) [inline kdl-client/graphql/schema.graphql:6]: wrong location, apiTokens should be located as an element of User. A server will have different users with different tokens. You should get your tokens using the `getMe` query.

File: kdl-client/src/Constants/routes.ts (hunk 1 above)
Comment (mgs95) [inline kdl-client/src/Constants/routes.ts:11]: make it plural:
```suggestion
  USER_API_TOKEN = '/server/:serverId/user/api-tokens',
```

File: kdl-client/src/Pages/Server/pages/Server/Server.tsx [hunk 4]
```diff
@@ -6,6 +6,7 @@ import Project from './pages/Project/Project';
 import Projects from './pages/Projects/Projects';
 import ServerBar from './components/ServerBar/ServerBar';
 import UserSshKey from './pages/UserSshKey/UserSshKey';
+import UserApiToken from './pages/UserApiToken/UserApiToken';
```
Comment (mgs95) [inline kdl-client/src/Pages/Server/pages/Server/Server.tsx:9]: Plural
```suggestion
import UserApiTokens from './pages/UserApiTokens/UserApiTokens';
```

File: kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.module.scss [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    background-color: palette(lowlight, 900);
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    .infoMessage {
+      @include font-body;
+      color: font-color(lowlight);
+    }
+    .generateButton {
+      margin-right: 3.5 * $grid-unit;
```
Comment (mgs95) [inline kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.module.scss:28]: change to: `margin: 0 3.5 * $grid-unit;`

File: kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.module.scss [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      margin-left: 3.5 * $grid-unit;
```
Comment (mgs95) [inline kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.module.scss:29]:
```suggestion
      margin: 0 3.5 * $grid-unit;
```

File: kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import styles from './UserApiToken.module.scss';
+import Token from './components/token/Token';
+import { GetApiTokens } from 'Graphql/queries/types/GetApiTokens';
+import Message from 'Components/Message/Message';
+
+const GetApiTokensQuery = loader('Graphql/queries/getApiTokens.graphql');
+
+function UserApiToken() {
+  const { data, loading, error } = useQuery<GetApiTokens>(GetApiTokensQuery);
+  const apiTokens = data?.apiTokens || [];
```
Comment (mgs95) [inline kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.tsx:15]: Unnecessary here. If there are no data API tokens will not be used, code will stop at l20. Check the following change:

File: kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.tsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  const isEmptyList = apiTokens.length === 0;
```
Comment (mgs95) [inline kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.tsx:16]:
```suggestion

```

File: kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+  function renderMainContent() {
+    if (loading) return <SpinnerCircular />;
+    else if (error) return <ErrorMessage />;
+    else if (isEmptyList) return <Message text="There are not tokens yet" />;
+
+    return apiTokens.map(({ label, lastUsedDate, creationDate }) => (
```
Comment (mgs95) [inline kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.tsx:23]:
```suggestion
    if (loading) return <SpinnerCircular />;
    if (error || !data) return <ErrorMessage />;
    
    if (!data.apiTokens.length) return <Message text="There are not tokens yet" />;

    return data.apiTokens.map(({ label, lastUsedDate, creationDate }) => (
```

File: kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.tsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    <div className={styles.container}>
+      <h1 className={styles.title}>API Tokens</h1>
+      <div className={styles.infoContainer}>
+        <p className={styles.infoMessage}>
+          This is your private SSH key, works lorem ipsum and for dolor amet. If
+          you want more information please lorem.
+        </p>
+        <Button
+          label="GENERATE"
+          border
```
Comment (mgs95) [inline kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.tsx:43]: move boolean props to bottom

File: kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.tsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+          className={styles.generateButton}
+          height={30}
```
Comment (mgs95) [inline kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiToken.tsx:45]:
```suggestion
          label="GENERATE"
          className={styles.generateButton}
          height={30}
          border
```

File: kdl-client/graphql/schema.graphql (hunk 3 above)
Comment (marco-gattelli) [reply kdl-client/graphql/schema.graphql:6]: 👍

File: kdl-client/graphql/schema.graphql (hunk 2 above)
Comment (marco-gattelli) [reply kdl-client/graphql/schema.graphql:19]: 👍

File: kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiTokens.tsx [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+const GetMeQuery = loader('Graphql/queries/getMe.graphql');
+
+function UserApiTokens() {
+  const { data, loading, error } = useQuery<GetMe>(GetMeQuery);
+
+  function handleGenerateClick() {
+    console.log('generate');
+  }
+  function handleDeleteClick(tokenId: string) {
+    console.log('delete', tokenId);
```
Comment (mgs95) [inline kdl-client/src/Pages/Server/pages/Server/pages/UserApiToken/UserApiTokens.tsx:19]: at this point you can perform this mutation, but lets complete in the next task

**Pull request merged**